cardano-node/src/Cardano/CLI/Genesis.hs

```diff
@@ -0,0 +1,71 @@
+{-# LANGUAGE CPP                        #-}
+{-# LANGUAGE RecordWildCards            #-}
+
+module Cardano.CLI.Genesis
+  ( dumpGenesis
+  )
+where
+
+import           Prelude (String)
+import           Cardano.Prelude hiding (option, show, trace)
+
+import qualified Data.ByteString.Lazy as LB
+import qualified Data.Map.Strict as Map
+import           Text.Printf (printf)
+
+import           System.Directory (createDirectory, doesPathExist)
+import           System.FilePath ((</>))
+#ifdef UNIX
+import           System.Posix.Files (ownerReadMode, setFileMode)
+#else
+import           System.Directory (emptyPermissions, readable, setPermissions)
+#endif
+
+import           Cardano.Chain.Delegation hiding (Map, epoch)
+import           Cardano.Chain.Genesis
+import           Cardano.Crypto (SigningKey (..))
+import qualified Cardano.Crypto.Signing as Crypto
+
+import           Cardano.CLI.Key
+import           Cardano.CLI.Ops
+
+
+dumpGenesis :: CLIOps IO -> FilePath -> GenesisData -> GeneratedSecrets -> IO ()
+dumpGenesis CLIOps{..} outDir genesisData GeneratedSecrets{..} = do
+  exists <- doesPathExist outDir
+  if exists
+    then throwIO $ OutputMustNotAlreadyExist outDir
+    else createDirectory outDir
+
+  let genesisJSONFile = outDir <> "/genesis.json"
+  LB.writeFile genesisJSONFile =<< coSerialiseGenesis genesisData
+
+  let dlgCertMap = unGenesisDelegation $ gdHeavyDelegation genesisData
```

Comment:
This needs a type signature.

Reply:
This is a code motion PR, Jordan!  It's intended to be quick, so it doesn't block others.